metadata-db-app/src/cmr/metadata_db/migrations/080_add_generic_document_table.clj

```diff
@@ -0,0 +1,74 @@
+(ns cmr.metadata-db.migrations.080-add-generic-document-table
+  (:require
+   [clojure.string :as string]
+   [config.mdb-migrate-helper :as h]))
+
+(defn- create-generic-documents-table
+  []
+  (h/sql
+   (string/replace
+    (str "CREATE TABLE METADATA_DB.cmr_generic_documents (
+          id NUMBER,
+          concept_id VARCHAR(255) NOT NULL,
+          native_id VARCHAR(1030) NOT NULL,
+          provider_id VARCHAR(10) NOT NULL,
+          document_name VARCHAR(20) NOT NULL,
+          schema VARCHAR(255) NOT NULL,
+          format VARCHAR(255) NOT NULL,
+          mime_type VARCHAR(255) NOT NULL,
+          metadata BLOB NOT NULL,
+          revision_id INTEGER DEFAULT 1 NOT NULL,
+          revision_date TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL,
+          created_at TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL,
+          deleted INTEGER DEFAULT 0 NOT NULL,
+          user_id VARCHAR(30),
+          transaction_id INTEGER DEFAULT 0 NOT NULL,
+
+          CONSTRAINT generic_doc_pk PRIMARY KEY (id),
+
+          CONSTRAINT gen_doc_cid_rev UNIQUE (concept_id, revision_id)
+          USING INDEX (create unique index generic_cri
+          ON cmr_generic_documents (concept_id, revision_id)))")
+    #"\s+" " ")))
+
+(defn- create-generic-document-indices
+  []
+  ;; Supports queries to find generic documents that have been deleted
+  (h/sql
+   (string/replace
+    (str "CREATE INDEX generic_documents_crdi
+          ON METADATA_DB.cmr_generic_documents (concept_id, deleted)")
+    #"\s+" " "))
+
+  ;; Supports queries to find generic document by native id within one provider
+  (h/sql
```

Comment:
This index can be removed. It is implied by the `(provider_id, native_id, revision_id)` index on the next statement.